storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/dataaccess/V4FinalizedKvStoreDao.java

```diff
@@ -116,6 +122,54 @@ public void close() throws Exception {
     return db.get(schema.getColumnNonCanonicalBlocksByRoot(), root);
   }
 
+  @Override
+  public void ingest(
+      final KvStoreFinalizedDao finalizedDao, final int batchSize, final Consumer<String> logger) {
+    Preconditions.checkArgument(batchSize > 1, "Batch size must be greater than 1 element");
+    Preconditions.checkArgument(finalizedDao instanceof V4FinalizedKvStoreDao);
+
+    final Map<String, KvStoreVariable<?>> newVariables = schema.getVariableMap();
+    if (newVariables.size() > 0) {
+      final Map<String, KvStoreVariable<?>> oldVariables =
+          ((V4FinalizedKvStoreDao) finalizedDao).schema.getVariableMap();
+      try (V4FinalizedUpdater updater = new V4FinalizedUpdater(db, schema, UInt64.ONE)) {
+        for (String key : newVariables.keySet()) {
+          logger.accept(String.format("Copy variable %s", key));
+          finalizedDao
+              .getRawVariable(oldVariables.get(key))
+              .ifPresent(value -> updater.transaction.putRaw(newVariables.get(key), value));
+        }
+        updater.commit();
+      }
+    }
+    final Map<String, KvStoreColumn<?, ?>> newColumns = schema.getColumnMap();
+    if (newColumns.size() > 0) {
+      final Map<String, KvStoreColumn<?, ?>> oldColumns =
+          ((V4FinalizedKvStoreDao) finalizedDao).schema.getColumnMap();
+      for (String key : newColumns.keySet()) {
+        logger.accept(String.format("Copy column %s", key));
+        try (final Stream<ColumnEntry<Bytes, Bytes>> oldEntryStream =
+                finalizedDao.streamRawColumn(oldColumns.get(key));
+            FinalizedStoreBatchWriter batchWriter =
+                new FinalizedStoreBatchWriter(batchSize, logger, db, schema)) {
+          oldEntryStream.forEach(entry -> batchWriter.add(newColumns.get(key), entry));
+        }
+      }
+    }
+  }
+
+  @Override
+  public <T> Optional<Bytes> getRawVariable(final KvStoreVariable<T> var) {
+    return db.getRaw(var);
+  }
+
+  @Override
+  @MustBeClosed
+  public <K, V> Stream<ColumnEntry<Bytes, Bytes>> streamRawColumn(
+      final KvStoreColumn<K, V> kvStoreColumn) {
+    return db.streamRaw(kvStoreColumn);
+  }
```

Comment:
These probably don't need to be on the interface given we're already requiring the instance to be a `V4FinalizedKvStoreDao` - I suspect they could just be private methods.